### PR TITLE
improve typing for openfaas provider

### PR DESCRIPTION
**Pull request by morrme**
Added type annotations to openfaas provider. See #9708 

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---

### CODE REVIEW

**Comment by potiuk** [comment]
Some static check errors :(. I heartily recommend installing and running pre-commit (see STATIC_CHECKS.rst) :)

**Comment by morrme** [comment]
@potiuk I need to get that working for sure!

**Review by mik-laj — APPROVED** [review]

File: airflow/providers/openfaas/hooks/openfaas.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def get_conn(self):
         conn = self.get_connection(self.conn_id)
         return conn
 
-    def deploy_function(self, overwrite_function_if_exist, body):
+    def deploy_function(self, overwrite_function_if_exist: bool, body: str) -> None:
```
Comment (mik-laj) [inline airflow/providers/openfaas/hooks/openfaas.py]:
```suggestion
    def deploy_function(self, overwrite_function_if_exist: bool, body: Dict[str, Any]) -> None:
```

File: airflow/providers/openfaas/hooks/openfaas.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def deploy_function(self, overwrite_function_if_exist, body):
             else:
                 self.log.info("Function deployed %s", self.function_name)
 
-    def invoke_async_function(self, body):
+    def invoke_async_function(self, body: str) -> None:
```
Comment (mik-laj) [inline airflow/providers/openfaas/hooks/openfaas.py]:
```suggestion
    def invoke_async_function(self, body: Dict[str, Any]) -> None:
```

File: airflow/providers/openfaas/hooks/openfaas.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ def invoke_async_function(self, body):
             self.log.error("Response status %d", response.status_code)
             raise AirflowException('failed to invoke function')
 
-    def update_function(self, body):
+    def update_function(self, body: str) -> None:
```
Comment (mik-laj) [inline airflow/providers/openfaas/hooks/openfaas.py]:
```suggestion
    def update_function(self, body: Dict[str, Any]) -> None:
```

**Comment by morrme** [comment]
@mik-laj Thanks!

**Pull request merged**